src/main/template/vpc.yaml

```diff
@@ -0,0 +1,442 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: This template deploys a VPC. Its IP range is quartered into four parts. The first three parts form private
+  subnets, the fourth is again quartered into three public subnets. Both the private and the public subnets are then
+  spread across three Availability Zones. All the required infrastructure, like internet gateway, nat gateways, and
+  default routes are included. Also, a bastion jump host is deployed in order to make any private machines reachable.
+
+Parameters:
+  cidrBlock:
+    Description: The CIDR range of the VPC. Note that it cannot be larger than 16 bits.
+    Type: String
+    Default: 10.0.0.0/16
+
+  firstQuarteringPartitionBits:
```

Comment:
we should rename this from "partition bits" (a made-up term?) to "cidr bits" (the name of the `Fn:Cidr` parameter we use this value for).

also, do we really need to allow different values for the individual quarters? this feels YAGNI to me.

Reply:
Note that those aren't for individual quarters but for the first and second quartering calcuation, where the second one results in eighths of the whole. This is why we need different values for both.
Your misinterpretation is a good example for the fact that the "user" of this template might not know what is expected of him. In fact, he can't choose freely. `cidrBlock`, `firstQuarteringPartitionBits` and `secondQuarteringPartitionBits` _have_ adhere to the correct mathematics, otherwise weird stuff will happen.
Perhaps, the best way is indeed to omit those parameters altogether and implement the arithmetic operations via a Lambda and a custom resource, like my comment above hints toward.
I simply was not sure if we need to add the complexity for that _now_ or if we can refactor that later.

Reply:
Implementing arithmetics via the lambda approach is too much complexity for now as the use case "but I need to tweak my private IP ranges arbitrarily" is YAGNI.

I think improving the clarity via naming and descriptions is sufficient; see my other comments.